sources/sqlserver/data.go

```diff
@@ -0,0 +1,263 @@
+		// We consider timezone for timestamp datatype.
+		// If timezone is not specified in mysqldump, we consider UTC time.
+		if TimezoneOffset == "" {
+			TimezoneOffset = "+00:00"
+		}
+		// convert timestamp from format "2006-01-02 15:04:05" to
+		// "2006-01-02T15:04:05+00:00".
+		timeNew := strings.Split(val, " ")
+		timeJoined := strings.Join(timeNew, "T")
+		timeJoined = timeJoined + TimezoneOffset
+		t, err = time.Parse(time.RFC3339, timeJoined)
+	} else {
+		// datetime: data should just consist of date and time.
+		// timestamp conversion should ignore timezone. We mimic this using Parse
+		// i.e. treat it as UTC, so it will be stored 'as-is' in Spanner.
+		t, err = time.Parse("2006-01-02 15:04:05", val)
+	}
+	if err != nil {
+		return t, fmt.Errorf("can't convert to timestamp (mysql type: %s)", srcTypeName)
+	}
+	return t, err
+}
+
+// convArray converts a source database string value (representing an
+// array) to an appropriate Spanner array value. It is the caller's
+// responsibility to detect and handle the case where the entire array
+// is NULL. However, convArray does handle the case where individual
+// array elements are NULL. In other words, convArray handles "{1,
+// NULL, 2}", but it does not handle "NULL" (it returns error).
+func convArray(spannerType ddl.Type, srcTypeName string, v string) (interface{}, error) {
+	v = strings.TrimSpace(v)
+	// Handle empty array. Note that we use an empty NullString array
+	// for all Spanner array types since this will be converted to the
+	// appropriate type by the Spanner client.
+	if v == "" {
+		return []spanner.NullString{}, nil
+	}
+
+	a := strings.Split(v, ",")
+
+	// The Spanner client for go does not accept []interface{} for arrays.
+	// Instead it only accepts slices of a specific type eg: []string
+	// Hence we have to do the following case analysis.
+	// NOTE: MySQL only supports SET of string which will be translated
+	// to spanner array<string>.
+	switch spannerType.Name {
+	case ddl.String:
+		var r []spanner.NullString
+		for _, s := range a {
+			if s == "NULL" {
+				r = append(r, spanner.NullString{Valid: false})
+				continue
+			}
+			s, err := processQuote(s)
+			if err != nil {
+				return []spanner.NullString{}, err
+			}
+			r = append(r, spanner.NullString{StringVal: s, Valid: true})
+		}
+		return r, nil
+	}
+	return []interface{}{}, fmt.Errorf("array type conversion not implemented for type %v", spannerType.Name)
+}
+
+// processQuote returns the unquoted version of s.
+// Note: The element values of PostgreSQL arrays may have double
+// quotes around them.  The array output routine will put double
+// quotes around element values if they are empty strings, contain
+// curly braces, delimiter characters, double quotes, backslashes, or
+// white space, or match the word NULL. Double quotes and backslashes
+// embedded in element values will be backslash-escaped.  See section
+// 8.14.6.of www.postgresql.org/docs/9.1/arrays.html.
```

Comment:
This is postgres specific, should not be here ?

Reply:
Sure, we have already removed those will be fixed in the next PR update.